src/StoreApi/Utilities/ProductQuery.php

```diff
@@ -317,6 +317,9 @@ public function add_query_clauses( $args, $wp_query ) {
 		if ( $wp_query->get( 'stock_status' ) ) {
 			$args['join']   = $this->append_product_sorting_table_join( $args['join'] );
 			$args['where'] .= ' AND wc_product_meta_lookup.stock_status IN ("' . implode( '","', array_map( 'esc_sql', $wp_query->get( 'stock_status' ) ) ) . '")';
+		} elseif ( 'yes' === get_option( 'woocommerce_hide_out_of_stock_items' ) ) {
+			$args['join']   = $this->append_product_sorting_table_join( $args['join'] );
+			$args['where'] .= ' AND wc_product_meta_lookup.stock_status NOT IN ("outofstock")';
```

Comment:
> If we do want to add this as a block attribute, I'd argue that we should do it in a different PR rather than here as it would significantly increase the scope and complexity of this PR. What do you think?

That seems like a fair approach. I trust your judgement here. I'd suggest creating a followup issue though to prevent it getting forgotten if the PR can't be done right away.